raspberry/bzzz/sensors/pressure_sensor.py

```diff
@@ -14,197 +15,139 @@ def _getShort(data, index):
     :param index: The starting index in the data array.
     :return: A signed 16-bit integer value.
     """
-    return c_short((data[index] << 8) + data[index + 1]).value
-
-
-def _getUshort(data, index):
-    """
-    Extract an unsigned 16-bit value from a data array starting at the specified index.
-    :param data: A list or tuple containing byte data.
-    :param index: The starting index in the data array.
-    :return: An unsigned 16-bit integer value.
-    """
-    return (data[index] << 8) + data[index + 1]
-
-class PressureSensor:
-    def __init__(self, 
-                 data_processor, 
-                 window_length, 
-                 DEVICE_ADDRESS=0x77, 
-                 reference_pressure_at_sea_level=None, 
-                 log_file=None):
-        
+    value = data[index] * 256 + data[index + 1]
+    return value - 65536 if value > 32767 else value
+class BMP180Sensor:
+    def __init__(self, data_processor, window_length, DEVICE_ADDRESS=0x77, reference_pressure_at_sea_level=101325, log_file=None):
         self.daemon = True
         self.__lock = Lock()
-        self.__thread = Thread(target=self.__get_measurements_in_background_t, 
-                               args=[DEVICE_ADDRESS,reference_pressure_at_sea_level])
-        # Sensor parameters
+        self.__thread = Thread(target=self.__get_measurements_in_background)
         self.window_length = window_length
         self.DEVICE_ADDRESS = DEVICE_ADDRESS
-        self.smbus = smbus.SMBus(1)
-        self.data_processor = data_processor  # Data processor object
+        self.bus = smbus.SMBus(1)
+        self.data_processor = data_processor
         self.reference_pressure_at_sea_level = reference_pressure_at_sea_level
-
-        # Data attributes
         self.__current_pressure = None
         self.__pressure_readings_list = []
         self.__current_altitude = None
-
-        # Logger
         self.log_file = log_file
         if log_file is not None:
             feature_names = ("Date_Time", "Pressure", "Altitude")
-            self.logger = DataLogger(num_features=2, 
-                                     feature_names=feature_names)
-        
-        # Sensor initialization
+            self.logger = DataLogger(num_features=2, feature_names=feature_names)
         self.__thread.start()
-
-    def __get_measurements_in_background_t(self, DEVICE_ADDRESS, reference_pressure_at_sea_level):
+    def __get_measurements_in_background(self):
         while True:
-            for _ in range(self.window_length):
-                self.__pressure_readings_list.append(self._get_current_pressure_measurement())
-            if self.reference_pressure_at_sea_level is None:
-                self.reference_pressure_at_sea_level = self.data_processor.process(self.__pressure_readings_list)
-        
-            self.update_altitude()
-        # Write data into the log file
+            self._update_pressure()
```

Comment:
Why is there a **single** underscore at the beginning of this method? If it's a private method, it should start with `__`. If not, there shouldn't be an underscore.